topgun/k8s/k8s_suite_test.go

```diff
@@ -200,16 +334,15 @@ func getPods(namespace string, flags ...string) []pod {
 }
 
 func isPodReady(p pod) bool {
-	total := len(p.Status.ContainerStatuses)
-	actual := 0
-
-	for _, containerStatus := range p.Status.ContainerStatuses {
-		if containerStatus.Ready {
-			actual++
+	for _, condition := range p.Status.Conditions {
+		if condition.Type != "ContainersReady" {
```

Comment:
by looking at `containersready` rather than "running", we ensure that we're only counting as "ready" something that's truly ready (i.e., pass the readiness probe).

Reply:
just referenced https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions good read! good addition!